scripts/evaluate_squad_model.py

```diff
@@ -0,0 +1,188 @@
+import squad_eval
+
+from allennlp.common import Params
+from allennlp.common.params import replace_none
+from allennlp.common.checks import ConfigurationError
+from allennlp.data import DataIterator, DatasetReader, Vocabulary
+from allennlp.models import Model
+from allennlp.nn.util import arrays_to_variables, device_mapping
+
+
+def main(config_file):
+    config = Params.from_file(config_file)
+    dataset_reader = DatasetReader.from_params(config['dataset_reader'])
+    iterator_params = config['iterator']
+    iterator_keys = list(iterator_params.keys())
+    for key in iterator_keys:
+        if key != 'batch_size':
+            del iterator_params[key]
+    iterator_params['type'] = 'basic'
+    iterator = DataIterator.from_params(iterator_params)
+    evaluation_data_path = config['validation_data_path']
+
+    expected_version = '1.1'
+    with open(evaluation_data_path) as dataset_file:
+        dataset_json = json.load(dataset_file)
+        if (dataset_json['version'] != expected_version):
+            print('Evaluation expects v-' + expected_version +
+                  ', but got dataset with v-' + dataset_json['version'],
+                  file=sys.stderr)
+        official_script_dataset = dataset_json['data']
+
+    cuda_device = 0
+    squad_eval.verbosity = 1
+    model = Model.load(config, cuda_device=cuda_device)
+
+    # Load the evaluation data
+    print("Reading evaluation data from %s" % evaluation_data_path)
+    dataset = dataset_reader.read(evaluation_data_path)
+    dataset.index_instances(model._vocab)
+
+    model.eval()
+    generator = iterator(dataset, num_epochs=1, shuffle=False)
+    print("Predicting best spans for the evalualtion data")
+    best_spans = []
+    result_dict = {}
+    for batch in tqdm.tqdm(generator):
+        tensor_batch = arrays_to_variables(batch, cuda_device, for_training=False)
+        result = model.forward(**tensor_batch)
+        best_span_tensor = result['best_span']
+        for i in range(best_span_tensor.size(0)):
+            best_spans.append(best_span_tensor[i].data.cpu().tolist())
+    for best_span, instance in zip(best_spans, dataset.instances):
+        span_tokens = instance.fields()['passage'].tokens()[best_span[0]:best_span[1]]
+        # We have to do some hacks to get from our tokens back to the original passage text, so
+        # that our answers get scored correctly.  This could be made much easier if we kept around
+        # the character offset in the original text when we tokenize things.
+        span_text = fix_span_text(span_tokens, instance.metadata['original_passage'])
+        question_id = instance.metadata['id']
+        result_dict[question_id] = span_text
+    metrics = model.get_metrics()
+    official_result = squad_eval.evaluate(official_script_dataset, result_dict)
+    print("Our model's metrics:", metrics)
+    print("Official result:", official_result)
+
+
+def fix_span_text(span_tokens, passage):
```

Comment:
This is arguably more complicated than just writing a reader for squad data which yields the pairs and classifying them individually using the inference function. What was the problem you foresaw with that, before I do it for the SRL model?